### ISSUE TITLE
Update dependency tsconfig-paths

### ISSUE THREAD

**Issue by Miljoen** [import-js/eslint-plugin-import#2639]
A high severity vulnerability in [JSON5](https://github.com/advisories/GHSA-9c47-m6qq-7p4h) was discovered.

`tsconfig-paths` relies on this package.

Fortunately, a [fix](https://github.com/dividab/tsconfig-paths/commit/9721a98718c9f6a35bb4029292204aaa90474bab) was created for [v4.1.2](https://github.com/dividab/tsconfig-paths/releases/tag/v4.1.2).

Request:

Update dependency 
```
"tsconfig-paths": "^3.14.1"
```

To

```
"tsconfig-paths": "^4.1.2"
```

**Comment by Teascade** [comment]
Bump. This is very annoying for me personally because the place I work at, this dependency is actually pretty much at the bottom of our dependency hierarchy so it's impossible to get rid of, and also because `npm audit` lacks a feature to disable specific packages from auditing. The only way for us to currently work around this is to literally change our automatic audits to `--audit-level critical` or `--omit=dev`, neither of which is good or optimal.

**Comment by air2** [comment]
You can use the eslint-plugins-i package

**Comment by air2** [comment]
Omitting CVE is a bad idea, because you could still trigger the CVE in JSON5 with a different route. So if an other package would depend on json5 the CVE could be very real, although through this package it will not be triggered. So only for this dependency tree it is a false positive.
Also since a few days the CVE is fixed in the v1 of json5 but the CVE db is not yet updated so it is now a real false positive.

**Comment by Teascade** [comment]
> You can use the eslint-plugins-i package

We can't, because we're also using `eslint-config-airbnb-base` which peer-depends on this package specifically.

**Comment by desaisagar** [comment]
I have created a [fix](https://github.com/import-js/eslint-plugin-import/pull/2641). Can someone review and merge it?

**Comment by air2** [comment]
@Teascade I use .npmrc with legacy-peer-deps=true to get around this issue. In my package.json I have: "eslint-plugin-import": "npm:eslint-plugin-i@^2.26.0-2",

**Comment by air2** [comment]
@desaisagar The maintainer of this package unfortunatly does not consider this a bug and will not merge fixes for it. (See all the other related and closed tickets)

**Comment by desaisagar** [comment]
Okay, thanks for the information @air2.

**Comment by gustaff-weldon** [comment]
@air2 any idea why maintainer is not concerned in helping to fix a CVE security issue?

**Comment by cbruchwaldnetfonds** [comment]
@gustaff-weldon you may find the explanation in this issue https://github.com/import-js/eslint-plugin-import/issues/2631

**Comment by air2** [comment]
It is because it is not a CVE for this package and ask everybody just to ignore this CVE, which is of course a bad idea, because the CVE could also be triggered if another packages uses json5. Also he wants to support node4, because not supporting node4 would be a breaking change, and needs a major version increment, which would be more work for him.

**Comment by Teascade** [comment]
> @Teascade I use .npmrc with legacy-peer-deps=true to get around this issue. In my package.json I have: "eslint-plugin-import": "npm:eslint-plugin-i@^2.26.0-2",

I would as well, but for some weird reason this results in an error `Cannot read property 'spec' of undefined` from `npm` itself with some projects, not all. Probably a bug, but makes the feature completely unusable for me.

**Comment by ljharb** [comment]
json5 v1.0.2 has already been updated with this fix, and either way, it's *not a valid vulnerability*.

As is the case with almost every JS CVE, the best course of action is to *do nothing* until the ecosystem fixes it for you.

This is a duplicate of #2625; a duplicate of #2628; a duplicate of #2626; a duplicate of #2627; a duplicate of #2631; a duplicate of #2632; a duplicate of #2634; a duplicate of #2635; a duplicate of #2636; a duplicate of #2637.

Please stop filing issues about a vulnerability on "not the vulnerable package", it doesn't help.

**Comment by mashpie** [comment]
@ljharb please stop telling to ignore CVE per se!

**Comment by ljharb** [comment]
@mashpie why? it's a good one to ignore, and spreading FUD about it doesn't make anything more secure.

**Comment by mashpie** [comment]
@ljharb with your Package possibly  not vulnerable doesn‘t mean there won‘t be any exploits for this CVE of json5.

ignoring it will ignore all vulnerable attack vectors beside of your package too. Like: chromiim, tap, nuxt, air-bnb, some pm2 etc.

json5 maintainer (@jordabtucker) himself committed to this CVE to be risky and mitigated it in all Majors. 

So: Ok, assuming you package won‘t exploit that vulnerabilty, others might do so. Telling your community to ignore this CVE and generalizing it in way like „… As is the case with almost every JS CVE…“ is wrong, missleading and dangerous for the whole ecosystem you - on the other side - relay on to „… until the ecosystem fixes it for you…“

please don‘t get ne wrong: Fixes ARE applied by Json5 in v1 and v2 and it‘s not eslint-Plugin-Import to be blamed. Story finished. Keep CVEs serious and to investigated and mitigated rather than to be ignored. 

Thanks! Have a nice 2023.

**Comment by mashpie** [comment]
…sry for spelling: autocorrection on mobile ;)

**Comment by air2** [comment]
It was a real CVE until json5 was upgraded, even because it was not triggered by this package does not say it was not a real CVE. If you are ignoring a CVE you ignore it for all packages that use this as dependecy. So if i have different dependecies that use json5 it could be a real CVE for me. Ignoring it would let me not see the CVE if i included a package in the future that uses json5 and was vulnerable. The only reason it can now be ignored is because it is fixed in json v1. But that was not the reason you told people to ignore it in the first place.

**Comment by ljharb** [comment]
@mashpie first of all, i have been very explicit in explaining that *this code path* isn't valid, but either way, Prototype Pollution is a class of CVEs that isn't actually that exploitable in practice - primarily only in theory.

If we all want CVEs to be considered seriously, then the entire system (one designed for boxed software and hardware) needs to be rethought from scratch for open source, which would include considering code paths, so that transitive warnings don't trigger unnecessarily - because in security, false negatives are MUCH WORSE than false positives, because they undermine confidence in the security systems themselves. The most dangerous thing for the ecosystem is treating every CVE as equally critical and scary.

**Comment by Miljoen** [comment]
@ljharb Hi man, I just wanted to thank you for the patient explanation given in https://github.com/import-js/eslint-plugin-import/issues/2631.

I've learned something about the npm ecosystem here.

And lastly, excuse us for making so many duplicate issues on your repo for what is, ultimately, out of your control.

See you!

**Comment by ljharb** [comment]
Thanks, I appreciate that.
